frontend/kubecloud/src/utils/userService.ts

```diff
@@ -112,7 +132,18 @@ export class UserService {
 
   // Reserve a node
   async reserveNode(nodeId: number, data: ReserveNodeRequest = {}) {
-    return api.post(`/v1/user/nodes/${nodeId}`, data, { requiresAuth: true, showNotifications: true })
+    const response = await api.post<ApiResponse<ReserveNodeResponse>>(`/v1/user/nodes/${nodeId}`, data, { requiresAuth: true, showNotifications: true })
+    const workflowChecker = createWorkflowStatusChecker(response.data.data.workflow_id, { interval: 6000 })
+    const status = await workflowChecker.status
+    if (status === WorkflowStatus.StatusFailed) {
+      useNotificationStore().addNotification({
+        title: 'Node Reservation Failed',
+        message: 'Failed to reserve node',
+        type: 'error',
+        duration: 5000
+      })
+    }
```

Comment:
In case of failure, it will return as well and have 2 notifications at the same moment

Reply:
Please check that on all other ewf used